client/azure/azure.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// Config contains the common attributes to create an Azure client.
+type Config struct {
+	// Dependencies
+	Logger micrologger.Logger
+
+	// ClientID is the ID of the Active Directory Service Principal.
```

Comment:
This is the config for the Active Directory Service Principal we need to use to authenticate against the Azure APIs.